### PR TITLE
libpiksi implementation and integration

### PR DESCRIPTION
**Pull request by jacobmcnamee**
- Refactor `libsbp_zmq` and `libsbp_settings` into `libpiksi`
- Significant API improvements
- Miscellaneous functional improvements
- Basic logging API based on `syslog`
- Update `sbp_settings_daemon`, `sbp_fileio_daemon`, and `piksi_system_daemon` to use `libpiksi`
- Doxygen documentation for `libpiksi` API

TODO:
- [x] Doxygen comments for API functions
- [x] Testing

/cc @swift-nav/firmware @fnoble

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-72-g74f669d
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-72-g74f669d/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-72-g74f669d/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-72-g74f669d

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-73-g151c688
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-73-g151c688/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-73-g151c688/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-73-g151c688

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-86-ge13b473
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-86-ge13b473/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-86-ge13b473/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-86-ge13b473

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-95-g9f549fc
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-95-g9f549fc/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-95-g9f549fc/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-95-g9f549fc

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-97-g326d1f8
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-97-g326d1f8/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-97-g326d1f8/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-97-g326d1f8

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-97-ge4f5c21
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-97-ge4f5c21/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-97-ge4f5c21/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-97-ge4f5c21

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-99-gc1e9a9f
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-99-gc1e9a9f/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-99-gc1e9a9f/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-99-gc1e9a9f

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-100-g56a2306
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-100-g56a2306/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-100-g56a2306/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-100-g56a2306

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-100-gd109216
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-100-gd109216/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-100-gd109216/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-100-gd109216

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-108-g2e9e630
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-108-g2e9e630/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-108-g2e9e630/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-108-g2e9e630

File: package/piksi_system_daemon/src/main.c [hunk 1]
```diff
@@ -141,60 +116,26 @@ static int uart_configure(const uart_t *uart)
...
-    printf("error configuring tty\n");
+    piksi_log(LOG_ERR, "error configuring tty");
     return -1;
   }
 
   return 0;
 }
 
-static bool baudrate_notify(struct setting *s, const char *val)
+static int baudrate_notify(void *context)
```
Comment (anth-cole) [inline package/piksi_system_daemon/src/main.c]: nit: `baudrate_notify` and `flow_control_notify` have the same signature and seem to do exactly the same thing, is it worth consolidating to one function maybe `uart_notify`?

File: package/sbp_settings_daemon/src/settings.c [hunk 2]
```diff
@@ -184,11 +184,11 @@ static void settings_read_reply_callback(u16 sender_id, u8 len, u8 msg[], void *
   const char *section = NULL, *setting = NULL, *value = NULL;
 
   if(!settings_parse_setting(len, msg, &section, &setting, &value, NULL))
-    log_error("Error in register message");
+    piksi_log(LOG_WARNING, "Error in register message");
```
Comment (anth-cole) [inline package/sbp_settings_daemon/src/settings.c]: did you mean to take these from error's to warnings? Seems like there's a lot of errors becoming warnings here, just checking it was deliberate

**Comment by wltr** [comment]
I agree with @anth-cole. I'm not familiar with the ins and outs of zmq and buildroot, but the code looks great to me. The comments would be helpful though.

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-122-gc00bc33
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-122-gc00bc33/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-122-gc00bc33/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-122-gc00bc33

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-122-g10984f7
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-122-g10984f7/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-122-g10984f7/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-122-g10984f7

**Pull request merged**